### ISSUE TITLE
Xunit reports formated incorrectly

### ISSUE THREAD

**Issue by jreading** [testem/testem#815]
The commit 284cfb3830cfba14fa9c7dd0a5c2b66a9b7229c5 by @rwjblue  is using the error node when it should be using the failure node. I don't think we are making the distinction between a failed test and an errored test. Consequently, failing tests aren't read correctly by Jenkins reader.

**Comment by rwjblue** [comment]
https://github.com/testem/testem/pull/816 reverts to using `<failure>` instead of `<error>` which I still personally think is incorrect, but I totally agree that a breaking change should be avoided.

https://github.com/testem/testem/commit/284cfb3830cfba14fa9c7dd0a5c2b66a9b7229c5 also properly logged actual failures that do not trigger errors (which were previously completely ignored), that portion of the commit is not reverted here.

**Comment by jreading** [comment]
Ok, I've tested this with various permutations of the XML with Jenkin's picky Junit reader. 

Everything seems to be parsing correctly, with `error` and `failure` nodes. 

I'm super sorry about the churn. It seems that your original PR is golden. I owe you all a beer. 
:beers: 

I saw failing tests not getting reported and your recent change and made a false connection. Good catch on the passed flag. Turns out we were only every reporting erroring tests, and I thought we were good. Time for a bike ride...
